bin/templates/scripts/cordova/lib/build.js

```diff
@@ -199,33 +219,52 @@ module.exports.findXCodeProjectIn = findXCodeProjectIn;
  * @param  {Boolean} isDevice      Flag that specify target for package (device/emulator)
  * @return {Array}                 Array of arguments that could be passed directly to spawn method
  */
-function getXcodeBuildArgs(projectName, projectPath, configuration, isDevice) {
+function getXcodeBuildArgs(projectName, projectPath, configuration, isDevice, buildFlags) {
     var xcodebuildArgs;
+    var options;
+    var buildActions = [ 'build' ];
+    var settings;
+    var customArgs = {};
+    customArgs.otherFlags = [];
+
+    if (buildFlags) {
+        if (typeof buildFlags === 'string' || buildFlags instanceof String) {
+            parseBuildFlag(buildFlags, customArgs);
+        } else { // buildFlags is an Array of strings
+            buildFlags.forEach( function(flag) {
+                parseBuildFlag(flag, customArgs);
+            });
+        }
+    }
+    
     if (isDevice) {
-        xcodebuildArgs = [
-            '-xcconfig', path.join(__dirname, '..', 'build-' + configuration.toLowerCase() + '.xcconfig'),
-            '-workspace', projectName + '.xcworkspace',
-            '-scheme', projectName,
-            '-configuration', configuration,
-            '-destination', 'generic/platform=iOS',
-            '-archivePath', projectName + '.xcarchive',
-            'archive',
```

Comment:
The automated tests do run the build command on Travis CI `npm run e2e-tests`

There is no check whether the archive artifact was created though since we can't really build for device on Travis